### PR TITLE
Refs #36219 - allow loading HashWithIndifferentAccess from YAML

### PR DESCRIPTION
**Pull request by evgeni**
Fixes: c9b82a573f64fe54e61d55f079adeb1df3d0cae8

### CODE REVIEW

File: app/models/report.rb [hunk 1]
```diff
@@ -62,7 +62,7 @@ def self.inherited(child)
   # extracts serialized metrics and keep them as a hash_with_indifferent_access
   def metrics
     return {} if self[:metrics].nil?
-    YAML.safe_load(read_metrics).with_indifferent_access
+    YAML.safe_load(read_metrics, permitted_classes: [ActiveSupport::HashWithIndifferentAccess]).with_indifferent_access
```
Comment (evgeni) [inline app/models/report.rb:65]: The alternative solution would be to make `read_metrics` not return HWIA, but I wasn't sure the mangling in that function is the only way for the data to get HWIA header, so this seemed safer.

**Comment by ofedoren** [comment]
A dumb question: since these errors are hard to catch before they come, isn't it better to define permitted classes as a constant and use it everywhere we use `safe_load`?.. Something like https://github.com/ofedoren/foreman/blob/develop/config/application.rb#L181

**Comment by evgeni** [comment]
I guess? I was first aiming at fixing the nightly issue (which it does, even if I don't understand why the existing tests didn't catch it before).

**Comment by evgeni** [comment]
[test katello]

**Review by ekohl — APPROVED** [review]
> A dumb question: since these errors are hard to catch before they come, isn't it better to define permitted classes as a constant and use it everywhere we use `safe_load`?.. Something like https://github.com/ofedoren/foreman/blob/develop/config/application.rb#L181

I think in general you want to limit it as much as possible.

**Comment by ekohl** [comment]
Not sure if this Katello failure is related. I wouldn't think it is.
```
[2023-07-31T19:03:20.150Z] Cancelling nested steps due to timeout
[2023-07-31T19:03:20.244Z] Sending interrupt signal to process
[2023-07-31T19:03:25.704Z] AccessPermissionsTest#test_0799_route foreman_tasks/api/tasks/callback should have a permission that grants access rake aborted!
[2023-07-31T19:03:25.704Z] SignalException: SIGTERM
```

**Comment by evgeni** [comment]
almost sure it isn't

[test katello]

**Comment by ekohl** [comment]
Doesn't look like it, merging.

**Pull request merged**